powerauth-push-client/src/main/java/io/getlime/push/client/PushServerClient.java

```diff
@@ -152,6 +164,21 @@ public boolean deleteDevice(Long appId, String token) throws PushServerClientExc
     }
 
     /**
+     * Update activation status for given device registration.
+     *
+     * @param activationId Identifier of activation
+     * @return True if updating went successful, false otherwise.
+     */
+    public boolean updateDeviceStatus(String activationId) throws PushServerClientException {
+        UpdateDeviceStatusRequest request = new UpdateDeviceStatusRequest();
+        request.setActivationId(activationId);
+        TypeReference<Response> typeReference = new TypeReference<Response>() {
+        };
+        ObjectResponse<?> response = putObjectImpl("/push/device/status/update", request, typeReference);
```

Comment:
Device status update must be `POST` - this method is called by the core of PowerAuth Server as an activation status change callback, it cannot be changed to `PUT`.

Reply:
This has to be changed in the controller (in annotation) as well - we have discussed this and the method must be POST despite the method name.